src/background/services/accounts/handlers/avalanche_deleteAccounts.ts

```diff
@@ -0,0 +1,235 @@
+import { injectable } from 'tsyringe';
+import { ethErrors } from 'eth-rpc-errors';
+
+import { DAppRequestHandler } from '@src/background/connections/dAppConnection/DAppRequestHandler';
+import {
+  DAppProviderRequest,
+  type JsonRpcRequestParams,
+} from '@src/background/connections/dAppConnection/models';
+import { DEFERRED_RESPONSE } from '@src/background/connections/middlewares/models';
+import { openApprovalWindow } from '@src/background/runtime/openApprovalWindow';
+import { canSkipApproval } from '@src/utils/canSkipApproval';
+
+import { AccountsService } from '../AccountsService';
+import type { Action } from '../../actions/models';
+import type { ImportedAccount, PrimaryAccount } from '../models';
+import { isPrimaryAccount } from '../utils/typeGuards';
+import { SecretsService } from '../../secrets/SecretsService';
+
+type PrimaryWalletAccounts = {
+  [walletId: string]: PrimaryAccount[];
+};
+
+export type DeleteAccountsDisplayData = {
+  primary: PrimaryWalletAccounts;
+  imported: ImportedAccount[];
+  wallet: {
+    [walletId: string]: string;
+  };
+};
+
+type Params = [accountIds: string[]];
+
+@injectable()
+export class AvalancheDeleteAccountsHandler extends DAppRequestHandler<
```

Comment:
Can you also remove the [`accounts/handlers/deleteAccounts.ts`](https://github.com/ava-labs/core-extension/blob/17924c76cc62eb23d9a6b6e8ed70446b89c8ea86/src/background/services/accounts/handlers/deleteAccounts.ts#L13) handler and make `AccountsProvider` use your new handler instead? 🙂

We already got rid of the old `renameAccount.ts` when we were adding the [`AvalancheRenameAccountHandler.ts`](https://github.com/ava-labs/core-extension/blob/85cbf294315230af73e94e77b8cc439049844de0/src/background/services/accounts/handlers/avalanche_renameAccount.ts#L20) so Core Web could use it too and we now have just one handler which Extension's UI also uses:

https://github.com/ava-labs/core-extension/blob/85cbf294315230af73e94e77b8cc439049844de0/src/contexts/AccountsProvider.tsx#L107-L115


We could benefit from the logic & guards you added here (i.e. the old handler does not check if only the last accounts are being removed 🫣).